vm/devices/storage/disk_nvme/nvme_driver/src/driver.rs

```diff
@@ -680,10 +945,100 @@ impl<T: DeviceBacking> DriverWorkerTask<T> {
             cpu,
         })
     }
+
+    /// Wrapper around save() to consume its response (for AsyncRun thread).
+    async fn save_wrapper(&mut self, worker_state: &mut WorkerState) -> NvmeDriverSavedState {
+        match self.save(worker_state).await {
+            Ok(state) => state,
+            Err(_) => {
+                NvmeDriverSavedState {
```

Comment:
I'm a little confused: I take this to mean: if this ```save``` fails then ```save_wrapper``` will simply return invalid save state.

How does that integrate with higher layers?

Reply:
Good point. I think a better way will be to return Option<> or Result<> to the upper level so we can discard the whole saved state if there were errors.